### PR TITLE
🐛 Nikto: Support v2.5.0 format

### PR DESCRIPTION
**Pull request by manuel-sommer**
see issue #9274

### CODE REVIEW

**Comment by mtesauro** [comment]
@manuel-sommer Hmm. I think it's better to report at info and let people move it up to Low (or whatever) based on how they rate issues internally. That's probably better then potentially inflating the severity of a finding.

**Review by mtesauro — APPROVED** [review]
Approved.

I'm a fan of making the parser smarter so it handles multiple formats from a tool so thanks for doing that. Much better than YASTP (Yet Another Same Tool Parser)

**Comment by manuel-sommer** [comment]
> @manuel-sommer Hmm. I think it's better to report at info and let people move it up to Low (or whatever) based on how they rate issues internally. That's probably better then potentially inflating the severity of a finding.

This could also be a usecase for this: https://github.com/DefectDojo/django-DefectDojo/issues/9250
If you give me a hint on how to advance the api and parse the value there to the finding parser, I could make a PR and advance nikto in a way to have:
- default setting = severity Info
- api setting = severity from api

File: dojo/tools/nikto/parser.py [hunk 1]
```diff
@@ -43,26 +43,34 @@ def get_findings(self, filename, test):
...
-
+        if len(data) == 1 and type(data) is list:
+            data = data[0]
         dupes = dict()
         host = data.get("host")
         port = data.get("port")
         if port is not None:
             port = int(port)
         for vulnerability in data.get("vulnerabilities", []):
+            if vulnerability.get('OSVDB') is None:
```
Comment (cneill) [inline dojo/tools/nikto/parser.py:54]: Shouldn't this condition be flipped?
```suggestion
            if vulnerability.get('OSVDB') is not None:
```

**Review by cneill — CHANGES_REQUESTED** [review]

File: dojo/tools/nikto/parser.py [hunk 2]
```diff
@@ -43,26 +43,34 @@ def get_findings(self, filename, test):
...
+                    f"**msg:** `{vulnerability.get('msg')}`",
+                    f"**HTTP Method:** `{vulnerability.get('method')}`",
+                    f"**OSVDB:** `{vulnerability.get('OSVDB')}`",
+                ])
+            else:
+                description = "\n".join([
+                    f"**id:** `{vulnerability.get('id')}`",
+                    f"**msg:** `{vulnerability.get('msg')}`",
+                    f"**HTTP Method:** `{vulnerability.get('method')}`",
+                ])
```
Comment (Maffooch) [inline dojo/tools/nikto/parser.py:66]:
```suggestion
            description = "\n".join([
                f"**id:** `{vulnerability.get('id')}`",
                f"**msg:** `{vulnerability.get('msg')}`",
                f"**HTTP Method:** `{vulnerability.get('method')}`",
            ])
            if vulnerability.get('OSVDB') is not None:
                description += "\n" + f"**OSVDB:** `{vulnerability.get('OSVDB')}`"
```

**Review by Maffooch — CHANGES_REQUESTED** [review]
Reducing duplicated code

**Review by Maffooch — APPROVED** [review]
Thanks @manuel-sommer :)

**Review by cneill — APPROVED** [review]

**Review by blakeaowens — APPROVED** [review]

**Pull request merged**